dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/vitals/CPUVitalReader.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.vitals
+
+import com.datadog.android.core.internal.persistence.file.canReadSafe
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import java.io.File
+
+/**
+ * Reads the CPU `utime` based on the `/proc/self/stat` file.
+ * cf. documentation https://man7.org/linux/man-pages/man5/procfs.5.html
+ */
+internal class CPUVitalReader(
+    private val statFile: File = STAT_FILE
+) : VitalReader {
+
+    override fun readVitalData(): Double? {
+        if (!(statFile.existsSafe() && statFile.canReadSafe())) {
+            return null
+        }
+
+        val stat = statFile.readText()
+        val tokens = stat.split(' ')
+        return if (tokens.size > UTIME_IDX) {
+            tokens[UTIME_IDX].toDoubleOrNull()
```

Comment:
shouldn't we check for `0` here as well? because I'm not 100% sure that read from this system file is not broken on some exotic OEM. wdyt?

Reply:
Do you mean checking if the returned value is `0.0` ? This file is part of linux, not Android, so I'd be really surprised if some OEMs modified things this low level

Reply:
yes, this is just a thought. Unlikely that will happen, but having 0 may lead to a wrong data in the dashboard, insights, etc.